giraffe/src/components/geo/processing/tableProcessing.ts

```diff
@@ -0,0 +1,85 @@
+// Libraries
+import {keys, indexOf} from 'lodash'
+
+// Types
+import {GeoTable} from './GeoTable'
+import {isPivotSensible, PivotedGeoTable} from './PivotedGeoTable'
+import {NativeGeoTable} from './NativeGeoTable'
+import {Table} from '../../../types'
+
+// Utils
+import {EmptyGeoTable} from './EmptyGeoTable'
+
+// Constants
+export const FIELD_COLUMN = '_field'
+export const VALUE_COLUMN = '_value'
+export const TABLE_COLUMN = 'table'
+export const QUERY_META_COLUMNS = [
+  FIELD_COLUMN,
+  VALUE_COLUMN,
+  TABLE_COLUMN,
+  '_start',
+  '_stop',
+]
+
+export const TIME_COLUMN = '_time'
+export const START_COLUMN = '_start'
+export const STOP_COLUMN = '_stop'
+export const LON_COLUMN = 'lon'
+export const LAT_COLUMN = 'lat'
+export const GEO_HASH_COLUMN = 's2_cell_id'
+
+export const preprocessData = (
+  table: Table,
+  rowLimit: number,
+  autoPivoting: boolean,
+  onFinalTable: (table: GeoTable) => void
+): GeoTable => {
+  if (autoPivoting && isPivotSensible(table)) {
+    // don't delay rendering with data calculation
+    setTimeout(() => {
+      onFinalTable(new PivotedGeoTable(table, rowLimit))
+    }, 0)
+    return new EmptyGeoTable()
+  }
+  return new NativeGeoTable(table, rowLimit)
+}
+
+export const getColumnNames = (
+  table: Table,
+  autoPivoting: boolean
+): string[] => {
+  if (autoPivoting) {
+    return getFieldColumnValues(table)
+  }
+  return getNumericColumns(table)
+}
+
+const getFieldColumnValues = (table: Table): string[] => {
+  const fieldColumn = table.getColumn(FIELD_COLUMN, 'string')
+  const valueColumn = table.getColumn(VALUE_COLUMN)
+  if (!fieldColumn) return []
+  const entriesCount = fieldColumn.length
+  const fieldNames = {}
+  for (let i = 0; i < entriesCount; i++) {
+    const fieldName = fieldColumn[i]
+    if (typeof valueColumn[i] === 'number') fieldNames[fieldName] = true
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.